Refactor/move previous year content

ShugKnight24 opened the pull request:
- Add new route (/previous-events) for Previous events
  - Created a simple component that has the NavBar & Footer and will render passed in components
  - Added 2024 & 2023 Sessions and Speakers w/ a basic description
    - Updated 2023 speaker time data to work with `date-fns` dependency 
  - Happy to update the page content or the route
  - Added a comment to update NavBar once Aaron's NavBar PR is merged
    - Not sure if better to update the current navbar or create a separate one due to the page differences

- Removed 2024 content from the Home page
  - Update Sessions and Speaker section components with call to speakers link if no speakers data is passed in
    - Can remove or make into reusable subcomponent if that's desired 

- Closes #17 

Happy to update this as desired 

## Home Page Preview

<img width="1567" height="4635" alt="update_home_page" src="https://github.com/user-attachments/assets/dd79973e-d29b-4d08-ba2c-8725692f6270" />

## Previous Events route

<img width="1567" height="10327" alt="previous_events_page" src="https://github.com/user-attachments/assets/235f7277-820e-414d-8d7a-2d7ee2b5be6e" />

# Update 9/21
---

- Add a 404 page now that there are multiple routes that redirects the user to the home page after 5 seconds
- Update README for pages accordingly
- Closes #23 

## 404 Preview

<img width="970" height="493" alt="404_page" src="https://github.com/user-attachments/assets/181818bd-b92b-4b50-b764-3d7ad2c7e413" />

# Update 9/22

- Make SessionsSection collapsible
- Correct PropTypes
- Use defaultExpanded on SessionsSection and update usage on SpeakersSection components

## SessionsSection Collapsible Preview

## sessions expanded home page

<img width="1251" height="383" alt="sessions_25_expanded" src="https://github.com/user-attachments/assets/57bd3f40-e124-452d-893e-fc5d47930ab2" />

## sessions collapsed home page

<img width="1258" height="342" alt="sessions_25_collapsed" src="https://github.com/user-attachments/assets/f9776f83-c322-4984-907a-63dc15626ef7" />

## sessions collapsed previous events page

<img width="1256" height="578" alt="previous_event_sessions_collapsed" src="https://github.com/user-attachments/assets/5bcf692b-93ee-42bb-a23e-40b82335a5ed" />

ShugKnight24 committed:
refactor: Create route for previous event years and add prior year data

- Create a simple layout component with the NavBar & Footer
- Update 2023 speakers time to work properly with `date-fns` dependency
- Add comments to the Navbar

ShugKnight24 committed:
refactor: Remove 2024 data from home page & and add call to speakers

- Update Sessions and Speakers with call to speakers link if no speakers 
data

ShugKnight24 committed:
feat: Add a NotFound page that redirects the user to home page

- Update README
- Closes #23

JustinBeaudry reviewed (commented):
LGTM, but let's wait to merge until we get Aaron's PR merged in.

ShugKnight24 committed:
refactor: Make `SessionsSection` collapsible

- Correct `SessionsSection` PropTypes
- Correct `defaultExpanded` usage
- Add `defaultExpanded` to `SessionsSection` components

ShugKnight24 commented:
Thanks @JustinBeaudry, that's perfect - I also just added making the SessionsSections collapsible here as well

JustinBeaudry committed:
Merge branch '2025' into refactor/move_previous_year_content

JustinBeaudry committed:
Merge branch '2025' into refactor/move_previous_year_content

ShugKnight24 committed:
fix: Resolve merge conflicts

- Merge branch '2025' into refactor/move_previous_year_content

ShugKnight24 committed:
chore: Remove duplicate keys from settings.json
